src/main/java/net/finmath/time/TimeDiscretization.java

```diff
@@ -204,15 +218,39 @@ public int getTimeIndexNearestGreaterOrEqual(double time) {
 	}
 
 	@Override
-	public TimeDiscretizationInterface getTimeShiftedTimeDiscretization(double timeShift) {
-		double[] newTimeDiscretization = new double[timeDiscretization.length];
+    public TimeDiscretizationInterface getTimeShiftedTimeDiscretization(double timeShift) {
+        double[] newTimeDiscretization = new double[timeDiscretization.length];
 
-		for(int timeIndex=0; timeIndex<timeDiscretization.length; timeIndex++) {
-			newTimeDiscretization[timeIndex] = roundToTimeTickSize(timeDiscretization[timeIndex]+timeShift);
-		}
+        for (int timeIndex = 0; timeIndex < timeDiscretization.length; timeIndex++) {
+            newTimeDiscretization[timeIndex] = roundToTimeTickSize(timeDiscretization[timeIndex] + timeShift);
+        }
 
-		return new TimeDiscretization(newTimeDiscretization);
-	}
+        return new TimeDiscretization(newTimeDiscretization);
+    }
+
+    /**
+     * @param that Another time discretization containing points to add to the time discretization.
+     * @return A new time discretization containing both the time points of this and the other discretization.
+     */
+    @Override
+    public TimeDiscretizationInterface union(TimeDiscretizationInterface that) {
+        return new TimeDiscretization(
+                Stream.concat(Arrays.stream(timeDiscretization).boxed(), Arrays.stream(that.getAsDoubleArray()).boxed()),
+                Math.max(timeTickSize, that.getTickSize()));
```

Comment:
Why max? Better min? If max: two points which where different in A are no longer different in A union B, even if B is empty. If A is "rounded to daily" and B is "rounded to monthly" the union should be "rounded to daily".

Reply:
Not sure. Maybe like that: for intersect it is max. For union it is min. In both cases we have to have tickSize1 = integer * tickSize2